MSAL/src/MSALPublicClientApplication.m

```diff
@@ -744,15 +765,36 @@ - (void)acquireTokenForScopes:(NSArray<NSString *> *)scopes
     params.loginHint = loginHint;
     params.extraAuthorizeURLQueryParameters = extraQueryParameters;
     params.accountIdentifier = account.lookupAccountIdentifier;
-    params.validateAuthority = _validateAuthority;
-    params.extraURLQueryParameters = _sliceParameters;
-    params.tokenExpirationBuffer = _expirationBuffer;
-    params.extendedLifetimeEnabled = _extendedLifetimeEnabled;
-    params.clientCapabilities = _clientCapabilities;
+    
+    params.extraURLQueryParameters = _configuration.extraQueryParameters.extraURLQueryParameters;
 
+    NSMutableDictionary *extraAuthorizeURLQueryParameters = [_configuration.extraQueryParameters.extraAuthorizeURLQueryParameters mutableCopy];
+    [extraAuthorizeURLQueryParameters addEntriesFromDictionary:extraQueryParameters];
+    params.extraAuthorizeURLQueryParameters = extraAuthorizeURLQueryParameters;
+    params.extraTokenRequestParameters = _configuration.extraQueryParameters.extraTokenURLParameters;
+    
+    params.tokenExpirationBuffer = _configuration.tokenExpirationBuffer;
+    params.extendedLifetimeEnabled = _configuration.extendedLifetimeEnabled;
+    params.clientCapabilities = _configuration.clientApplicationCapabilities;
+
+    params.validateAuthority = _configuration.validateAuthority
+        || [MSIDAADAuthority isAuthorityFormatValid:_configuration.authority.url context:nil error:nil];
+    
+    if (params.validateAuthority && _configuration.knownAuthorities)
+    {
+        for (MSALAuthority *knownAuthority in _configuration.knownAuthorities)
+        {
+            if ([params.authority isKindOfClass:knownAuthority.class]
```

Comment:
where is params.authority getting set?

Reply:
In it's initializer? Id surely hope it gets set in initWithAuthority: method